llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Transforms/InParallelToHAL.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cstdlib>
+
+#include "iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.h"
+#include "iree-dialects/Dialect/LinalgExt/Transforms/Transforms.h"
+#include "iree-dialects/Dialect/LinalgExt/Transforms/Utils.h"
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+#include "mlir/Dialect/Arithmetic/IR/Arithmetic.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/SCF/SCF.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/IR/AffineExpr.h"
+#include "mlir/IR/BlockAndValueMapping.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/Operation.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "llvm/ADT/STLExtras.h"
+
+using namespace mlir;
+using namespace mlir::iree_compiler::IREE::LinalgExt;
+
+// TODO: This also needs to do the work of `SetNumWorkgroups` but we can't
+// depend on HAL atm.
+FailureOr<SmallVector<Operation *>> mlir::iree_compiler::IREE::LinalgExt::
```

Comment:
This pass needs lit tests as well. This part is load bearing to connect this in e2e. Without lit tests it is hard to see what this pass is expected to do. As things change in IREE core compiler, this part needs to be modified as well, and just e2e test just make it that much harder.

Reply:
right, the thinking is that this should go away but I can still add a test.

Reply:
Actually this one is not a pass but a rewrite that is introduced in the parent commit and tested via the transform dialect. This was already unit tested.